CS9057: 8.5.0 release does not build under .NET 8.14 (8.0.407)

### Things to check

- [x] I have searched the **existing issues** for this bug
- [x] To rule out a caching problem I made sure the bug also happens in an **incognito tab**

### Bug type

Other

### Component name

Analyzer

### What happened?

After bumping from 8.4.0 to 8.5.0 as per advice we get below error:

`CSC : error CS9057: The analyzer assembly 'C:\Users\VssAdministrator\.nuget\packages\mudblazor\8.5.0\analyzers\dotnet\cs\MudBlazor.Analyzers.dll' references version '4.13.0.0' of the compiler, which is newer than the currently running version '4.11.0.0'. [D:\a\1\s\src\CIM.Samples.BlazorWebApp\CIM.Samples.BlazorWebApp.csproj]`

Must be caused by bump of `Microsoft.CodeAnalysis.CSharp` to 4.13.0 in 8.5.0.

global.json
```
{
  "sdk": {
    "version": "8.0.407",
    "rollForward": "latestPatch"
  }
}
```

### Expected behavior

MudBlazor 8.5.0 works with .NET SDK 8.14.

### Reproduction link

https://github.com/perclausen/mudblazor-8.5.0-CS9057

### Reproduction steps

1. Reference MudBlazor 8.5.0
2. Try to build with .NET SDK 8.14 (8.0.407)


### Relevant log output

```shell

```

### Version (bug)

8.5.0

### Version (working)

8.4.0

### What browsers are you seeing the problem on?

Other

### On which operating systems are you experiencing the issue?

Windows

### Pull Request

- [ ] I would like to do a Pull Request

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

> > But it also fails in our Azure DevOps pipeline where VS isn't installed.
> 
> So far I see dotnet build also builds it?
> 
> ```
> C:\Users\Dell\Desktop\mudblazor-8.5.0-CS9057-main\src>dotnet --info
> .NET SDK:
>  Version:           8.0.407
>  Commit:            cac65f27eb
>  Workload version:  8.0.400-manifests.011c56c6
>  MSBuild version:   17.11.26+2b19be476
> 
> Runtime Environment:
>  OS Name:     Windows
>  OS Version:  10.0.19045
>  OS Platform: Windows
>  RID:         win-x64
>  Base Path:   C:\Program Files\dotnet\sdk\8.0.407\
> 
> global.json file:
>   C:\Users\Dell\Desktop\mudblazor-8.5.0-CS9057-main\global.json
> 
> Learn more:
>   https://aka.ms/dotnet/info
> 
> Download .NET:
>   https://aka.ms/dotnet/download
> 
> C:\Users\Dell\Desktop\mudblazor-8.5.0-CS9057-main\src>dotnet build
>   Determining projects to restore...
>   All projects are up-to-date for restore.
>   mudblazor-8.5.0-CS9057 -> C:\Users\Dell\Desktop\mudblazor-8.5.0-CS9057-main\src\bin\Debug\net8.0\mudblazor-8.5.0-CS90
>   57.dll
> 
> Build succeeded.
>     0 Warning(s)
>     0 Error(s)
> 
> Time Elapsed 00:00:04.75
> 
> Workload updates are available. Run `dotnet workload list` for more information.
> 
> C:\Users\Dell\Desktop\mudblazor-8.5.0-CS9057-main\src>
> ```

Results from our Azure DevOps pipeline where SDK installed using global.json.
```
- task: UseDotNet@2
  displayName: 'Use .NET SDK (global.json)'
  inputs:
    packageType: 'sdk'
    useGlobalJson: true
    workingDirectory: '$(System.DefaultWorkingDirectory)'
```

```
dotnet sdk check
.NET SDKs:
Version      Status     
------------------------
8.0.407      Up to date.

Try out the newest .NET SDK features with .NET 9.0.202.

.NET Runtimes:
Name                              Version      Status     
----------------------------------------------------------
Microsoft.AspNetCore.App          8.0.14       Up to date.
Microsoft.NETCore.App             8.0.14       Up to date.
Microsoft.WindowsDesktop.App      8.0.14       Up to date.
```

And the error:
```
CSC : error CS9057: The analyzer assembly 'C:\Users\VssAdministrator\.nuget\packages\mudblazor\8.5.0\analyzers\dotnet\cs\MudBlazor.Analyzers.dll' references version '4.13.0.0' of the compiler, which is newer than the currently running version '4.11.0.0'. [D:\a\1\s\src\CIM.Samples.BlazorWebApp\CIM.Samples.BlazorWebApp.csproj]
  CIM.Samples.BlazorWebApp -> D:\a\1\s\src\CIM.Samples.BlazorWebApp\bin\Release\net8.0\CIM.Samples.BlazorWebApp.dll
```

--- a/src/MudBlazor.Analyzers.TestComponents/MudBlazor.Analyzers.TestComponents.csproj
+++ b/src/MudBlazor.Analyzers.TestComponents/MudBlazor.Analyzers.TestComponents.csproj
@@ -8,6 +8,9 @@
 
   <ItemGroup>
     <ProjectReference Include="..\MudBlazor\MudBlazor.csproj" />
+	<!--https://github.com/MudBlazor/MudBlazor/issues/11114-->
+	<!--https://github.com/dotnet/roslyn/issues/77255-->
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="4.13.0" />
   </ItemGroup>
   
   <ItemGroup>

--- a/src/MudBlazor.Analyzers/MudBlazor.Analyzers.csproj
+++ b/src/MudBlazor.Analyzers/MudBlazor.Analyzers.csproj
@@ -8,7 +8,7 @@
   
   <ItemGroup>
     <PackageReference Include="Microsoft.CodeAnalysis.Analyzers" Version="3.11.0" />
-    <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="4.13.0" />
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="4.7.0" />
   </ItemGroup>
 
   <ItemGroup>

